File: Editorial/137_Single_Number_II/CPP/3_Mathematics.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_set<long> numsSet;
        long sumNums = 0;
        for (int num : nums) {
            numsSet.insert((long)num);
            sumNums += num;
        }

        long sumSet = 0;
        for (long num : numsSet) {
            sumSet += num;
        }

        return (int)((3 * sumSet - sumNums) / 2);
    }
};

File: Editorial/137_Single_Number_II/TypeScript/3_Mathematics.ts
function singleNumber(nums: number[]): number {
    let numsSet = new Set(nums);
    let sumNums = nums.reduce((a, b) => a + b, 0);
    let sumSet = Array.from(numsSet.values()).reduce((a, b) => a + b, 0);
    return (3 * sumSet - sumNums) / 2;
}
